s3/s3.go

```diff
@@ -221,6 +223,7 @@ func init() {
 // Constants
 const (
 	metaMtime      = "Mtime"                       // the meta key to store mtime in - eg X-Amz-Meta-Mtime
+	metaMD5Hash    = "Md5chksum"                   // the meta key to store md5hash in
```

Comment:
> I note in your branch you use metaMD5Hash = "Md5hash" - I'd like to follow some standard here if there is one! Any thoughts?

Other than S3's special-purpose `Content-MD5` field, which it keeps separate from the metadata dictionary, I don't think S3 has a standard MD5sum field [[1](http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html#object-metadata)].  The closest I have found is AWS's use of `md5chksum` in its documentation [[2](https://aws.amazon.com/premiumsupport/knowledge-center/data-integrity-s3/)].  I switched to using `md5chksum` here.